Components/Sensors/ThermocoupleTask.cpp

```diff
@@ -0,0 +1,331 @@
+		}
+		if(Error1 & 0x02){
+			SOAR_PRINT("Thermocouple 1 is shorted to GND\n");
+		}
+		if(Error1 & 0x04){
+			SOAR_PRINT("Thermocouple 1 is shorted to VCC\n");
+		}
+		SOAR_PRINT("\n");
+	}
+	else
+	{
+		SOAR_PRINT("Thermocouple 1 is reading %d.%d C \n\n" , temperature1/100, temperature1%100);
+	}
+
+	//thermo 2 print
+	if(dataBuffer2[1] & 0x01)
+	{								  // Returns Error Number
+		if(Error2 & 0x01){
+			SOAR_PRINT("Thermocouple 2 is not connected\n");
+		}
+		if(Error2 & 0x02){
+			SOAR_PRINT("Thermocouple 2 is shorted to GND\n");
+		}
+		if(Error2 & 0x04){
+			SOAR_PRINT("Thermocouple 2 is shorted to VCC\n");
+		}
+		SOAR_PRINT("\n");
+	}
+	else
+	{
+		SOAR_PRINT("Thermocouple 2 is reading %d.%d C \n\n", temperature2/100, temperature2%100);
+	}
+}
+
+/**
+ * @brief This method receives the voltage reading through spi from the thermocouple readings
+ */
+void ThermocoupleTask::SampleThermocouple()
+{
+	/*DATA FROM MAX31855KASA+T ------------------------------------------------------
+
+	32 bits Memory Map
+
+		D31-D18 : Thermocoupler Temperature Data
+
+			D31 : Sign bit
+
+			D30-D18 : Temperature Value (2's complement) from 2^10 to 2^-2
+
+		D17 : Reserved Bit
+
+		D16 : Fault (if high shows fault is detected, more specific fault messages at D2 - D0)
+
+		D15-D4 :  Internal Temperature Data (reference junction temperature)
+
+			D15 : Sign bit
+
+			D14-D4 : Temperature Value (2's complement) from 2^6 to 2^-4
+
+		D3 : Reserved
+
+		D2-D0 : Fault Detection Bits
+
+			D2 : SCV Fault (displays high if TC shorts to Vcc)
+
+			D1 : SCG Fault (displays high if TC shorts to GND)
+
+			D0 : Thermocouple has no Connection (displays high)
+
+	*///------------------------------------------------------------------------------
+
+	//Storable Data ------------------------------------------------------------------------------
+
+
+	SOAR_PRINT("\n-- Sample Thermocouple Data --\n");
+
+	uint8_t tempDataBuffer5[5] = {0};
+	//See Above bit mem-map
+
+	//Read ---------------------------------------------------------------------------------------
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_SET);
+	HAL_GPIO_WritePin(TC2_NCS_GPIO_Port, TC2_NCS_Pin, GPIO_PIN_SET);
+
+
+    //Read From Thermocouple 1 first
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_RESET); //begin read with CS pin low
+	HAL_Delay(10);
+	HAL_SPI_Receive(SystemHandles::SPI_Thermocouple1, tempDataBuffer5, 5, 1000); //Fill the data buffer with data from TC1
+	HAL_Delay(10);
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_SET); //end read with setting CS pin to high again
+
+	for(int i = 0; i<5; i++){
+		if(i!=4)
+		dataBuffer1[i] = tempDataBuffer5[i];\
+	}
+
+	int Temp1=0;
+
+	if(!(dataBuffer1[1] & 0x01)){ //if there is not an error with TC1 compute temperature
+		if((dataBuffer1[0]&(0x80))>>7==1)  // Negative Temperature
+		{
+			Temp1 = (dataBuffer1[0] << 6) | (dataBuffer1[1] >> 2);
+			Temp1&=0b01111111111111;
+			Temp1^=0b01111111111111;
```

Comment:
Maybe check just in case, but if it works thats ok.

Note: we googled seems to be an xor?